archunit/src/main/java/com/tngtech/archunit/library/cycle_detection/Cycle.java

```diff
@@ -31,11 +34,19 @@
  * @param <EDGE> The type of the edges forming the cycle
  */
 @PublicAPI(usage = ACCESS, state = EXPERIMENTAL)
-public interface Cycle<EDGE extends Edge<?>> {
+public interface Cycle<EDGE extends Edge<?>> extends Convertible {
 
     /**
      * @return The edges of the {@link Cycle}
      */
     @PublicAPI(usage = ACCESS, state = EXPERIMENTAL)
     List<EDGE> getEdges();
+
+    @Override
+    default <T> Set<T> convertTo(Class<T> type) {
+        return getEdges().stream()
+                .filter(edge -> edge instanceof Convertible)
+                .flatMap(edge -> ((Convertible) edge).convertTo(type).stream())
```

Comment:
I have no intuition whether it makes sense that a cycle of convertible edges can be converted to the combined conversion of those edges.

Do you have a (real life) example in mind?

Reply:
Yes, `Cycle` -> `Set<Dependency>`. I think that's fair, no? It could also be `Set<Set<Dependency>>`, but I don't think that's easier to handle for all practical purposes (which is usually something like "I want to visualize all dependencies that form a cycle")